util/flags/tls.go

```diff
@@ -0,0 +1,74 @@
+/*
+Copyright 2022 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// Package flags implements the webhook server TLS options utilities.
+package flags
+
+import (
+	"crypto/tls"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/pflag"
+	cliflag "k8s.io/component-base/cli/flag"
+)
+
+// SecureWebhookOptions has the options to configure the TLS settings for a webhook server.
+type SecureWebhookOptions struct {
+	TLSMinVersion   string
+	TLSCipherSuites []string
+}
+
+// AddSecureWebhookFlags adds the webhook server TLS configuration flags
+// to the default flag set.
+func AddSecureWebhookFlags(fs *pflag.FlagSet, options *SecureWebhookOptions) {
+	fs.StringVar(&options.TLSMinVersion, "tls-min-version", "",
+		"The minimum TLS version in use by the webhook server.\n"+
+			fmt.Sprintf("Possible values are %s.", strings.Join(cliflag.TLSPossibleVersions(), ", ")),
+	)
+
+	tlsCipherPreferredValues := cliflag.PreferredTLSCipherNames()
+	tlsCipherInsecureValues := cliflag.InsecureTLSCipherNames()
+	fs.StringSliceVar(&options.TLSCipherSuites, "tls-cipher-suites", []string{},
+		"Comma-separated list of cipher suites for the webhook server. "+
+			"If omitted, the default Go cipher suites will be used. \n"+
+			"Preferred values: "+strings.Join(tlsCipherPreferredValues, ", ")+". \n"+
+			"Insecure values: "+strings.Join(tlsCipherInsecureValues, ", ")+".")
+}
+
+// CalculateTLSOpts returns a list of TLS configuration overrides
+// to be used when setting the webhook server.
+func CalculateTLSOpts(options SecureWebhookOptions) ([]func(*tls.Config), error) {
```

Comment:
nitpicking on naming, feel free to say no:

What about `GetTLSConfigSetter` (because it returns a func that sets a tls.Config)

Reply:
How about `GetTLSOptionOverrideFuncs`??

Reply:
just for context. Controller-runtime called those funcs `TLSOpts` I'm fine either way

Reply:
+1 to GetTLSOptionOverrideFuncs

I saw Controller-runtime using TLSOpts, but I personally prefer to avoid abbreviation in public types/names + calling opts a func seems a little bit confusing, hence my suggestions